src/DIRAC/Resources/IdProvider/IdProviderFactory.py

```diff
@@ -1,19 +1,14 @@
-########################################################################
-# File :   IdProviderFactory.py
-# Author : A.T.
-########################################################################
-
 """  The Identity Provider Factory instantiates IdProvider objects
      according to their configuration
 """
 import jwt
 
-from DIRAC import S_OK, S_ERROR, gLogger
+from DIRAC import S_OK, S_ERROR, gLogger, gConfig
 from DIRAC.Core.Utilities import ObjectLoader, ThreadSafe
-from DIRAC.Core.Utilities.DictCache import DictCache
-from DIRAC.Resources.IdProvider.Utilities import getProviderInfo, getSettingsNamesForIdPIssuer
 from DIRAC.FrameworkSystem.private.authorization.utils.Clients import getDIRACClients
 from DIRAC.FrameworkSystem.private.authorization.utils.Utilities import collectMetadata
+from DIRAC.Resources.IdProvider.Utilities import getIdProviderIdentifierFromIssuerAndClientID
+
 
 gCacheMetadata = ThreadSafe.Synchronizer()
```

Comment:
If you remove the cacheMetadata which is not currently used then you should remove this line as well together with the corresponding import